### ISSUE TITLE
Janitor's gear teleported around the station from his inventory

### ISSUE THREAD

**Issue by ironsnake345** [tgstation/tgstation#49925]
## Reproduction: Screw around with the wielded push broom enough?

We were playing as box station, traitor round, and I spawned in as Janitor, non-antag, and loaded up my kit to get to work. Some time into the round, I think a bit over an hour, after I've leveled my scrubbing skill to legendary, I became a syndicate sleeper agent, and was assigned to assassinate the detective, steal the medal of captaincy, and escape on the shuttle. Shortly after I was antagged, I purchased some gear: an emag, an agent ID, and a chameleon kit. I ran to the showers and, frequently honking the rubber duck inside as I moved around, started to change into my chameleon kit. I fumbled most parts of the kit and they went in and out of inventories and layers of inventories more times than I can count, but eventually I had everything on. After some moving around station, customizing my chameleon kit a bit more and stealing a botany tray and some chemicals, I disguised as the captain and tried to emag the shuttle console to recall the shuttle, but that didn't work. After that, I got back on the janicart, grabbed my janitorial cart once again, undisguised, bought a minibomb and a couple random items (including dart gun) and headed over to escape to catch the shuttle. There, at about 1 hour 30 minutes round time, a nightmare with envy's knife approached me and started slashing at the lights and APCs, then offered me the knife. I toggled my flashlight on and off a few times, and all of a sudden, my flashlight and tablet appeared on the ground beside me. There was no alert about the nightmare trying to take anything off me, and when I tried to pick it back up and put it back into my pockets, it acted like it had just been used inhand, like it was somehow codified as being  both in my pocket and in the hand that had just picked it up off the floor. Then my backpack and several items of my clothing (I think my shoes, gloves and hat) just vanished completely. I came to learn later that the clothing items had wound up in the hands of people on the other side of the station, while the minibomb and dart gun teleported directly  into the hands of people standing right beside me. The nightmare then proceeded to open fire with my dart gun while a random nearby scientist ran off with the minibomb. With nothing else left to do, I ran into medical and committed suicide with the janicart key. As soon as I died, all of a sudden my clothes popped back onto my body, and I think it was the original clothing I was wearing at the start of the shift - not the chameleon wear - but I'm not sure about that. I never got Envy's knife, and I don't think I was ever struck by it.

Update: Based on the comments, I'm pretty sure this was caused by some weird interaction with the push broom. I'm not entirely sure how I pulled it off, and I think it is one of the items that vanished from my inventory (my hands specifically) so I'm not sure how to recreate it. Either way, it caused some big time fuckery.

**Comment by ironsnake345** [comment]
I just remembered, I was toying around with the push broom and trash bag for a while to sweep up bullet shells and shards. I don't know if that changes anything, but I feel like it's worth mentioning.

**Comment by LemonInTheDark** [comment]
Could you move the main wall of text out of the header? It makes it quite hard to parse

**Comment by XDTM** [comment]
The broom had a bug that made it able to sweep items off mobs, without properly unhooking the inventory procs. So that's probably the cause here.

**Comment by ironsnake345** [comment]
I'll see if I can figure out how to move the wall of text out of the header. I will say, regarding the broom, at the time I may have been wielding it as a weapon. I think you're right about that being the cause. It might have been on the janitorial cart at the time, though. Either way, inventory procs being improperly unhooked sounds like what I was experiencing.

**Comment by LemonInTheDark** [comment]
I love fast fixes where I don't have to do work.

There's a chance this is something else, wait for swanni to take a look at this issue and the pr before closing it.

**Comment by ATH1909** [comment]
I wasn't a nightmare, I was a shadowperson.

**Comment by ironsnake345** [comment]
I frankly don't know the difference.
